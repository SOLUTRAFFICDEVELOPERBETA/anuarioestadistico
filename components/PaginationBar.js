import React from 'react'
import { Box, Typography, Grid, IconButton } from '@material-ui/core'
import { ChevronLeft, ChevronRight, SkipPrevious, SkipNext } from '@material-ui/icons'
import PropTypes from 'prop-types'

/**
 * Barra de pagina para la lista limitada
 * @param {*} props Propiedades del componente
 */
const PaginationBar = ({ onNext, onSkipNext, showNext, onBack, onSkipBack, showBack, label }) => (
  <Box>
    <Grid
      container
      alignContent="center"
      alignItems="center"
      justify="flex-end"
    >
      <Grid item xl={3} lg={3} md={3} sm={6} xs="auto">
        <Typography
          align="center"
          color="primary"
          variant="caption"
          display="block"
        >
          {label}
        </Typography>
      </Grid>
      <Grid item xl={2} lg={2} md={2} sm={6} xs="auto">
        <IconButton
          size="small"
          color="primary"
          onClick={onSkipBack}
          disabled={showBack}
        >
          <SkipPrevious />
        </IconButton>
        <IconButton
          size="small"
          color="primary"
          onClick={onBack}
          disabled={showBack}
        >
          <ChevronLeft />
        </IconButton>
        <IconButton
          size="small"
          color="primary"
          onClick={onNext}
          disabled={showNext}
        >
          <ChevronRight />
        </IconButton>
        <IconButton
          size="small"
          color="primary"
          onClick={onSkipNext}
          disabled={showNext}
        >
          <SkipNext />
        </IconButton>
      </Grid>
    </Grid>
  </Box>
)

PaginationBar.propTypes = {
  label: PropTypes.string,
  showNext: PropTypes.bool,
  showBack: PropTypes.bool,
  onNext: PropTypes.func.isRequired,
  onBack: PropTypes.func.isRequired,
  onSkipNext: PropTypes.func.isRequired,
  onSkipBack: PropTypes.func.isRequired
}

PaginationBar.defaultProps = {
  label: '',
  showBack: false,
  showNext: false
}
export default PaginationBar
