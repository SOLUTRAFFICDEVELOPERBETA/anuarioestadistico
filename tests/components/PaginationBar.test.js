import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import PaginationBar from '../../components/PaginationBar';

describe('Pruebas en <PaginationBar/>', () => {
    const onBack = jest.fn();
    const onSkipBack = jest.fn();
    const onNext = jest.fn();
    const onSkipNext = jest.fn();
    const wrapper = shallow(
        <PaginationBar
            onNext={onNext}
            onSkipNext={onSkipNext}
            showNext={true}
            onBack={onBack}
            onSkipBack={onSkipBack}
            showBack={true}
            label={`Pagina 1 de 4`}
        />
    )

    test('debe Mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe retornar el testo correctamente', () => {

        expect(wrapper.find('WithStyles(ForwardRef(Typography))').text().trim()).toBe('Pagina 1 de 4')
    })

    test('debe llamar la funciones de onBack(), onSkipBack(), onNext() y onSkipNext()', () => {
        wrapper.find('WithStyles(ForwardRef(IconButton))').at(0).simulate('click');
        wrapper.find('WithStyles(ForwardRef(IconButton))').at(1).simulate('click');
        wrapper.find('WithStyles(ForwardRef(IconButton))').at(2).simulate('click');
        wrapper.find('WithStyles(ForwardRef(IconButton))').at(3).simulate('click');

        expect(onSkipBack).toHaveBeenCalled();
        expect(onBack).toHaveBeenCalled();
        expect(onNext).toHaveBeenCalled();
        expect(onSkipNext).toHaveBeenCalled();

        expect(onSkipBack).toHaveBeenCalledTimes(1);
        expect(onBack).toHaveBeenCalledTimes(1);
        expect(onNext).toHaveBeenCalledTimes(1);
        expect(onSkipNext).toHaveBeenCalledTimes(1);

    })
})
