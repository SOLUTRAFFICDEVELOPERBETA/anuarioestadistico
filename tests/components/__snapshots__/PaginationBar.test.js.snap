// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PaginationBar/> debe Mostrarse correctamente 1`] = `
<Styled(MuiBox)>
  <WithStyles(ForwardRef(Grid))
    alignContent="center"
    alignItems="center"
    container={true}
    justify="flex-end"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={3}
      md={3}
      sm={6}
      xl={3}
      xs="auto"
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        color="primary"
        display="block"
        variant="caption"
      >
        Pagina 1 de 4
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={2}
      md={2}
      sm={6}
      xl={2}
      xs="auto"
    >
      <WithStyles(ForwardRef(IconButton))
        color="primary"
        disabled={true}
        onClick={[MockFunction]}
        size="small"
      >
        <SkipPreviousIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        color="primary"
        disabled={true}
        onClick={[MockFunction]}
        size="small"
      >
        <ChevronLeftIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        color="primary"
        disabled={true}
        onClick={[MockFunction]}
        size="small"
      >
        <ChevronRightIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        color="primary"
        disabled={true}
        onClick={[MockFunction]}
        size="small"
      >
        <SkipNextIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Styled(MuiBox)>
`;
